spec/lib/suse/connect/api_spec.rb

```diff
@@ -274,159 +254,67 @@
 
       context 'when sending in bulk' do
         let(:systems) { create_list :system, 3, :full, scc_synced_at: nil }
-        let(:product) { system.products.first }
-        let(:hw_info) { system.hw_info }
-
-        let(:expected_body) do
-          system_hashes = systems.collect do |system|
-            product = system.products.first.slice(*product_keys)
-            hwinfo = system.hw_info.slice(*hwinfo_keys)
-            {
-              login: system.login,
-              password: system.password,
-              last_seen_at: system.last_seen_at,
-              hostname: system.hostname,
-              hwinfo: hwinfo,
-              products: [product]
-            }
-          end.compact
-          { systems: system_hashes }
-        end
 
         it 'yields successful results' do
           expect(api_client.send_bulk_system_update(relation)).to eq(expected_response)
         end
       end
 
       context 'when sending in bulk with and without system_token' do
-        let!(:system1) { create :system, :full, :with_system_token, :synced }
-        let!(:system2) { create :system, :full, :synced }
-
+        let!(:system1) { create :system, :full, :with_system_token }
+        let!(:system2) { create :system, :full }
         let(:systems) { [system1, system2] }
 
-        let(:expected_body) do
-          {
-            systems: [
-              {
-                login: system1.login,
-                password: system1.password,
-                last_seen_at: system1.last_seen_at,
-                system_token: system1.id
-              },
-              {
-                login: system2.login,
-                password: system2.password,
-                last_seen_at: system2.last_seen_at
-              }
-            ]
-          }
-        end
-
         it 'yields successful results' do
           expect(api_client.send_bulk_system_update(relation)).to eq(expected_response)
         end
       end
 
       context 'when sending in bulk and encounter 413 http error code' do
```

Comment:
This test was broken and was green no matter what was changed. I improved the test to make sure it is actually calling the endpoint multiple times!